gestalt/__init__.py

```diff
@@ -155,6 +154,61 @@ def build_config(self) -> None:
         self.__conf_data = self.__flatten(self.__conf_data,
                                           sep=self.__delim_char)
 
+        self.__parse_keys()
+        self.__interpolate_keys()
+
+    def __parse_keys(self) -> None:
+        """Parses the keys in the configuration data.
+
+        Raises:
+            RuntimeError: If the configuration data is not valid
+        """
+        for k, v in self.__conf_data.items():
+            if not isinstance(v, str):
+                continue
+            m = self.regex_pattern.search(v)
+            if m is None:
+                continue
+            if m.group(1) not in self.__providers:
+                raise RuntimeError(
+                    "Provider not configured yet expect to be used")
+            if v in self.__secret_map:
+                self.__secret_map[v].append(k)
+            else:
+                self.__secret_map.update({v: [k]})
+
+    def configure_provider(self, provider_name: str,
+                           provider: Provider) -> None:
+        """Configures a provider for use in the library.
+
+        Args:
+            provider_name (str): The name of the provider to configure
+            provider (Provider): The provider to configure
+
+        Raises:
+            TypeError: If the provider is not an instance of the Provider class
+        """
+        if provider_name == "vault" and isinstance(provider, Vault):
+            self.__providers.update({"vault": provider})
+        else:
+            raise TypeError("Provider provider is not supported")
+
+    def __interpolate_keys(self) -> None:
+        """Interpolates the keys in the configuration data.
+        """
+        for path, v in self.__secret_map.items():
+            m = self.regex_pattern.search(path)
+            if m is not None:
+                provider = self.__providers[m.group(1)]
```

Comment:
What happens if someone tries to build with provider references, and that provider hasn't been declared / configured?